clmm/theory/parent_class.py

```diff
@@ -692,36 +712,57 @@ def eval_reduced_tangential_shear(self, r_proj, z_cl, z_src, z_src_model='single
         if self.validate_input:
             validate_argument(locals(), 'r_proj', 'float_array', argmin=0)
             validate_argument(locals(), 'z_cl', float, argmin=0)
-            validate_argument(locals(), 'z_src', 'float_array', argmin=0)
+            validate_argument(locals(), 'z_src_info', str)
+            validate_argument(locals(), 'approx', str, none_ok=True)
+            if z_src_info=='discrete':
+                validate_argument(locals(), 'z_src_info', 'float_array', argmin=0)
```

Comment:
Related to the comment on the docstrings, `'z_src_info'` would be `'z_src'`. The same applies to the next two `validate_argument`.